### ISSUE TITLE
 Chat - Console error and email changes to "a%20href=" for multiline email hyperlinks

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#43386]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**
**Reproducible in staging?:**
**Reproducible in production?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4602126
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:**
**Slack conversation:**

## Action Performed:
1. Log in to New Expensify
2. Navigate to any chat
3. Draft a message with multiline email hyperlink
4. Note a console error
5. Send the message
6. Edit the message and note email address
7. Cancel the edit, copy the message > paste it as plain text

## Expected Result:
There shouldn't be any console errors and the email address should remain the same as when drafting the message.
Just the text part of the hyperlink should be pasted when pasting as plain text

## Actual Result:
Console error related to react-native-live-markdown-parser is displayed and email address of the link changes to "a%20href=".
Additionally, the URL and the text is pasted when pasting multiline hyperlinks as plain text instead of just text
ERROR MESSAGE
react-native-live-markdown-parser.js:34 TypeError: Cannot read properties of undefined (reading 'children')
at react-native-live-markdown-parser.js:32:658
at Array.forEach (<anonymous>)
at react-native-live-markdown-parser.js:32:504
at globalThis.parseExpensiMarkToRanges (react-native-live-markdown-parser.js:32:3552)
at p (parserUtils.js:170:25)
at MarkdownTextInput.web.js:149:24
at MarkdownTextInput.web.js:259:16
at Object.De (react-dom.production.min.js:54:317)
at Le (react-dom.production.min.js:54:471)
at react-dom.production.min.js:55:35

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/e0f8e0d2-c196-4b58-b5d7-0a22f7faca26

[logs (2).txt](https://github.com/user-attachments/files/15775470/logs.2.txt)

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
@dylanexpensify  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp

**Comment by BartoszGrajdek** [comment]
Hi! I'm Bartosz from SWM one of the maintainers of `react-native-live-markdown` I'll try to take a look at this issue this week 👀

**Comment by dylanexpensify** [comment]
reproing today!

**Comment by dylanexpensify** [comment]
Thanks @BartoszGrajdek!

**Comment by BartoszGrajdek** [comment]
Added an in-depth update [here](https://github.com/Expensify/App/issues/43922#issuecomment-2180764904). 

I believe this is something that is happening with any kind of hyperlink (although it may not be the same error in the console log). Not sure yet if it's a duplicate or separate issue, so I'll keep investigating it 🙌🏻 

<details>
<summary>Video of multiline URL hyperlink</summary>


https://github.com/Expensify/App/assets/41169160/fd54d092-04b4-4c0c-a6b5-3b6b6468fefc



</details>

**Comment by hungvu193** [comment]
Thanks for your update 😄

**Comment by dylanexpensify** [comment]
Nice! What's our eta on merge?

**Comment by hungvu193** [comment]
Friendly bump @BartoszGrajdek

**Comment by BartoszGrajdek** [comment]
Hi! Sorry - I had a deploy blocker to resolve on Friday, so I didn't manage to get back to this issue, looking into it today! 👀

**Comment by BartoszGrajdek** [comment]
@thienlnam this is also an issue related to live markdown, could we add it to the project? 🙏🏻

**Comment by BartoszGrajdek** [comment]
After some investigation, I'm certain that this problem comes from `expensify-common` namely `ExpensiMark.js` and how it parses our markdown. Let me show you what's happening.
## Working Example
### Input
```
[test](email@gmail.com)
```
### Output
```
<a href="mailto:email@gmail.com" data-raw-href="email@gmail.com" data-link-variant="labeled">testtest</a>
```
## Our Issue with line breaks
### Input
```
[test
](email@gmail.com)
```
### Output
_(I tried formatting it a little, to make it more readable)_
```
<a 
    href="mailto:
        <a href="mailto:email@gmail.com" data-raw-href="email@gmail.com" data-link-variant="auto">
            email@gmail.com
        </a>" 
    data-raw-href="
        <a href="mailto:email@gmail.com" data-raw-href="email@gmail.com" data-link-variant="auto">
            email@gmail.com
        </a>" 
    data-link-variant="labeled"
>
    test
</a>
```

## Next steps
The fix will probably need to change regex [here](https://github.com/Expensify/expensify-common/blob/797090255f87e37f546f85165b40f83a41f0d3ee/lib/ExpensiMark.ts#L452-L457). Not sure how, but it somehow is adding `a` tags into `href` & `data-raw-href`.

If anyone could come up with a reasonable proposal I'm happy to give this task away to external contributors, supporting them along the way.

**Comment by hungvu193** [comment]
Thanks for your update @BartoszGrajdek.
But do we really need to support that kind of content? I mean, who gonna add a line break inside an email? I think with our current line break strategy, normal can't do that.

https://github.com/Expensify/App/assets/16502320/f3785f75-e526-4d54-9ce0-49c40e336476

**Comment by hungvu193** [comment]
> The line break is actually in the text of the hyperlink, not the email itself.

Yeah, this's what I mean. Let's me bump on slack to get more 👀

**Comment by hungvu193** [comment]
I'll review proposal today

**Comment by hungvu193** [comment]
@ShridharGoel Would you mind to explain a little about your solution? Thank you.

**Comment by hungvu193** [comment]
Same here. Still looking for proposals

**Comment by hungvu193** [comment]
Do you think we should raise the bounty for more 👀 here @dylanexpensify?

**Comment by dylanexpensify** [comment]
done!

**Comment by hungvu193** [comment]
I'll review today

**Comment by hungvu193** [comment]
Thanks @skyweb331, your proposal looks promising. However if we allowed `newline` for our `filterRules`, it will show the empty lines even when we don't have any character after that, please checkout this evidence.

https://github.com/Expensify/App/assets/16502320/64d7b269-0db9-4d13-b8d9-9ab702df7c15

**Comment by skyweb331** [comment]
@hungvu193 Thanks for your point-outs.
Will you let me know your inputs in the video? I could not get what you input to get that result...

**Comment by hungvu193** [comment]
Here you are:

```
[test












](test@gmail.com)
```

**Comment by skyweb331** [comment]
In this case, you want to remove all empty line breaks? so output should be just 
`<a href="mailto:test@gmail.com">test</a>`

**Comment by hungvu193** [comment]
> In this case, you want to remove all empty line breaks? so output should be just `<a href="mailto:test@gmail.com">test</a>`

I believe so, it's similar when we add few line break after a work then enter.

**Comment by skyweb331** [comment]
OK. I will fix my proposal now.

**Comment by skyweb331** [comment]
@hungvu193 Proposal updated and following additional test passes.

```javascript
    const multilineEmail = '[test\n\n\n](test@test.com)';
    const parsedEmail =
            '<a href="mailto:test@test.com">test</a>';

    expect(parser.replace(multilineEmail)).toBe(parsedEmail);
```

**Comment by skyweb331** [comment]
@hungvu193 Sorry, Some existing tests are failed. Will check again.

**Comment by skyweb331** [comment]
@hungvu193 Proposal updated and all tests passed.

**Comment by hungvu193** [comment]
This looking good. I'll try few more tests before making final review 😄

**Comment by hungvu193** [comment]
@skyweb331 's [proposal](https://github.com/Expensify/App/issues/43386#issuecomment-2220120810) looks good to me!
🎀 👀 🎀   C+ reviewed

**Comment by skyweb331** [comment]
PR is created : https://github.com/Expensify/expensify-common/pull/750

**Comment by skyweb331** [comment]
Not overdue. PR is merged.

**Comment by hungvu193** [comment]
PR is merged

**Comment by hungvu193** [comment]
@skyweb331 Feel free to raise a PR to update expensify-common and react-native-live-markdown in EApp.

**Comment by hungvu193** [comment]
I think the automation is failing here. This issue will be ready for payment on Jul, 30th

**Comment by hungvu193** [comment]
payment is due today cc @dylanexpensify

**Comment by JmillsExpensify** [comment]
Waiting on payment summary for approval.

**Comment by JmillsExpensify** [comment]
$500 approved for @hungvu193

**Comment by mallenexpensify** [comment]
@dylanexpensify I updated your payment summary because I paid @skyweb331.

**Comment by dylanexpensify** [comment]
ty @mallenexpensify ❤️ !
